common/common-client/src/main/java/org/ow2/proactive/authentication/iam/IAMRestClient.java

```diff
@@ -34,20 +34,36 @@
 import org.apache.http.HttpStatus;
 import org.apache.http.NameValuePair;
 import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.message.BasicNameValuePair;
 import org.apache.http.util.EntityUtils;
+import org.jasig.cas.client.util.CommonUtils;
 import org.jsoup.Jsoup;
 import org.jsoup.nodes.Document;
 import org.ow2.proactive.http.CommonHttpClientBuilder;
 
 
 public class IAMRestClient {
 
+    /**
+     * An attribute (key) used to generate SSO ticket as a JWT (Json Web Token)
+     */
+    private static final String TOKEN_KEY = "token";
+
+    // The attribute value set to 'true' when SSO ticket is generated as a JWT
+    private static final String TOKEN_VALUE = "true";
+
+    /**
+     * Pattern used to extract SSO ticket generated as a TGT (Ticket-Granting-Ticket)
+     * Example: TGT-1-cS39DsDtjIXUV4-zVF23dYys5GpfSIGPmWU7vdA8tzM0aI7FB4H52EG6BLPADMik3zA-activeeon
+     */
+    private static final Pattern pattern = Pattern.compile("(TGT-\\d+-\\S+)");
+
     private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(IAMRestClient.class.getName());
 
-    public String getSSOTicket(String url, String username, String password, String ticketMarker) {
+    public String getSSOTicket(String url, String username, String password, String ticketMarker, boolean asJWT) {
```

Comment:
unclear to me why the rest client should transform the sso ticket returned by the IAM. If the format of the ticket is meant to be configured globally on the iam server side.